src/main/java/org/opensearch/agent/tools/SearchAnomalyDetectorsTool.java

```diff
@@ -123,24 +131,92 @@ public <T> void run(Map<String, String> parameters, ActionListener<T> listener)
+
+                        // We follow the existing logic as the frontend to determine overall detector state
+                        // https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/server/routes/utils/adHelpers.ts#L437
+                        String detectorState;
+                        ADTask realtimeTask = profileResponse.getRealtimeAdTask();
+
+                        if (realtimeTask != null) {
+                            String taskState = realtimeTask.getState();
+                            if (taskState.equalsIgnoreCase("CREATED")) {
+                                detectorState = DetectorStateString.Initializing.name();
+                            } else if (taskState.equalsIgnoreCase("RUNNING")) {
+                                detectorState = DetectorStateString.Running.name();
+                            } else if (taskState.equalsIgnoreCase("INIT_FAILURE")
+                                || taskState.equalsIgnoreCase("UNEXPECTED_FAILURE")
+                                || taskState.equalsIgnoreCase("FAILED")) {
+                                detectorState = DetectorStateString.Failed.name();
+                            } else {
+                                // Task states may fall under other values, such as "FEATURE_REQUIRED" / "STOPPED" / etc.
+                                // Defaulting all other states to fall under the disabled category
+                                detectorState = DetectorStateString.Disabled.name();
+                            }
+                        } else {
+                            detectorState = DetectorStateString.Disabled.name();
+                        }
+
+                        if (running != null && !running && detectorState.equals(DetectorStateString.Running.name())) {
+                            hitsAsMap.remove(detectorId);
+                        }
+                        if (disabled != null && !disabled && detectorState.equals(DetectorStateString.Disabled.name())) {
+                            hitsAsMap.remove(detectorId);
+                        }
+                        if (failed != null && !failed && detectorState.equals(DetectorStateString.Failed.name())) {
+                            hitsAsMap.remove(detectorId);
+                        }
+                    }
+                }
+            }
+
             sb.append("AnomalyDetectors=[");
-            for (SearchHit hit : hits) {
+            for (SearchHit hit : hitsAsMap.values()) {
                 sb.append("{");
                 sb.append("id=").append(hit.getId()).append(",");
                 sb.append("name=").append(hit.getSourceAsMap().get("name"));
                 sb.append("}");
             }
             sb.append("]");
-            sb.append("TotalAnomalyDetectors=").append(response.getHits().getTotalHits().value);
+            sb.append("TotalAnomalyDetectors=").append(hitsAsMap.values().size());
```

Comment:
```suggestion
            sb.append("TotalAnomalyDetectors=").append(hitsAsMap.size());
```

Reply:
I could integrate the map removal as part of the above conditionals, but I prefer to leave as-is for more readability. I would need to leave in the extra else blocks as well.